.github/workflows/publish_github_package.yml

```diff
@@ -0,0 +1,43 @@
+name: Publish GitHub package
+
+on:
+  workflow_dispatch:
+  release:
+    types: [created]
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Use Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          registry-url: 'https://registry.npmjs.org'
+          cache: 'npm'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Set version to release tag
+        shell: bash
+        id: define_tag
+        run: |
+          PACKAGE_VERSION=$(git describe --abbrev=0 --tags | sed 's/^v//')
+          echo "PACKAGE_VERSION=$PACKAGE_VERSION" >> $GITHUB_OUTPUT
+
+      - name: Update package.json version
+        run: npm version ${{ steps.define_tag.outputs.PACKAGE_VERSION }} --no-git-tag-version
+
+      - name: Build
+        run: npm run build
```

Comment:
Aren't we suppose to build already before creating the release?

Reply:
No, we build before publish, but we need to set the version before build.